src/GameWorldSimulator/NeoServer.Game.Creatures/Models/Bases/CombatActor.cs

```diff
@@ -471,7 +486,9 @@ protected void InvokeAttackCanceled()
     public event BlockAttack OnBlockedAttack;
     public event Attack OnAttackEnemy;
     public event Damage OnInjured;
-    public event Die OnKilled;
+    public event PrepareDeath OnPrepareDeath;
```

Comment:
Can we rename to "OnCreatureDying" ?

Reply:
yes, fixed on last commit: https://github.com/OpenCoreMMO/OpenCoreMMO/pull/650/commits/860579bf572f65383870d8e11bc2596ff0d53822